traffic_ops/app/db/migrations/20171101000000_add_deep_caching_type.sql

```diff
@@ -0,0 +1,23 @@
+/*
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	    http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+-- +goose Up
+-- SQL in section 'Up' is executed when this migration is applied
+
+alter table deliveryservice add column deep_caching_type smallint default 0;
```

Comment:
any reason this couldn't be a bool if only 2 values? never and always. or are more values anticipated and in that case couldn't it be an enum? or whatever it's called in postgres. domain? (like signing_algorithm) in my experience, "magic numbers" always end up causing confusion/problems. for example, i can never remember what protocol 0, 1 and 2 stand for.

Reply:
Thanks for the suggestion.

I'm anticipating a 3rd value in the near future - `POPULAR` - where the client is only routed deep if the requested path is decidedly popular. Admittedly I did like the flexibility of this being just an int, which would make it easier to add different types in the future. However, I do agree with your concern about `protocol` for instance where the meaning of the numbers are really only known within the API.

I'll work on changing this int to an enum.